PriceKataCalc/Program.cs

```diff
@@ -47,10 +51,31 @@ public double calcPriceAfterTax()
             return this.price + this.calcTax();
         }
 
-        public double calcPriceAfterDiscount()
+        public double calcDiscount()
+        {
+            var myprice= calcPriceAfterTax() *this.discount;
+            return myprice;
+        }
+        public double calcUPCDiscount()
         {
-            return calcPriceAfterTax() * (1 - this.discount);
+            var myprice = this.price * UPCDiscount();
+            return myprice;
+        }
+        public double calcPriceAfterAllDiscount()
+        {
+            var myprice = this.price-calcDiscount()-this.calcUPCDiscount(); 
+            return myprice;
         }
 
+        private double UPCDiscount()
+        {
+            if (priceCalc.keysDiscount.ContainsKey(this.UPC)) return priceCalc.keysDiscount[this.UPC];
+            else return 0;
+        }
+
+        internal object calcAllDiscount()
```

Comment:
shouldn't be internal right ?

Reply:
i guess i forgot to edit after the refactoring tool